packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker/create-model.ts

```diff
@@ -0,0 +1,239 @@
+
+    // add the security groups to the connections object
+    if (props.vpcConfig) {
+      this.vpc = props.vpcConfig.vpc;
+      this.subnets = props.vpcConfig.subnets ? this.vpc.selectSubnets(props.vpcConfig.subnets).subnetIds : this.vpc.selectSubnets().subnetIds;
+    }
+
+    this.taskPolicies = this.makePolicyStatements();
+  }
+
+  /**
+   * The execution role for the Sagemaker Create Model API.
+   *
+   * Only available after task has been added to a state machine.
+   */
+  public get role(): iam.IRole {
+    if (this._role === undefined) {
+      throw new Error('role not available yet--use the object in a Task first');
+    }
+    return this._role;
+  }
+
+  public get grantPrincipal(): iam.IPrincipal {
+    if (this._grantPrincipal === undefined) {
+      throw new Error('Principal not available yet--use the object in a Task first');
+    }
+    return this._grantPrincipal;
+  }
+
+  /**
+   * Add the security group to all instances via the launch configuration
+   * security groups array.
+   *
+   * @param securityGroup: The security group to add
+   */
+  public addSecurityGroup(securityGroup: ec2.ISecurityGroup): void {
+    this.securityGroups.push(securityGroup);
+  }
+
+  /**
+   * @internal
+   */
+  protected _renderTask(): any {
+    return {
+      Resource: integrationResourceArn('sagemaker', 'createModel', this.integrationPattern),
+      Parameters: sfn.FieldUtils.renderObject(this.renderParameters()),
+    };
+  }
+
+  private renderParameters(): { [key: string]: any } {
+    return {
+      EnableNetworkIsolation: this.props.enableNetworkIsolation,
+      ExecutionRoleArn: this._role!.roleArn,
+      ModelName: this.props.modelName,
+      Tags: this.props.tags?.value,
+      PrimaryContainer: this.props.primaryContainer.bind(this).parameters,
+      Containers: this.props.containers?.map(container => (container.bind(this))),
+      ...this.renderVpcConfig(this.props.vpcConfig),
+    };
+  }
+
+  private makePolicyStatements(): iam.PolicyStatement[] {
+    const stack = cdk.Stack.of(this);
+    // set the SageMaker role or create new one
+    // https://docs.aws.amazon.com/sagemaker/latest/dg/sagemaker-roles.html
+    this._grantPrincipal = this._role =
+        this.props.role ||
+        new iam.Role(this, 'SagemakerRole', {
+          assumedBy: new iam.ServicePrincipal('sagemaker.amazonaws.com'),
+          inlinePolicies: {
+            CreateModel: new iam.PolicyDocument({
+              statements: [
+                new iam.PolicyStatement({
+                  actions: [
+                    'cloudwatch:PutMetricData',
+                    'logs:CreateLogStream',
+                    'logs:CreateLogGroup',
+                    'logs:PutLogEvents',
+                    'logs:DescribeLogStreams',
+                    'ecr:GetAuthorizationToken',
+                  ],
+                  resources: ['*'],
```

Comment:
for anything that uses `*`, let's add a comment to explain why they can't be scoped to a resource / link to documentation. It'll help in ensuring that we're keeping it to 'least privilege'

Reply:
Added links